### PR TITLE
fix(useVirtualList): get correct offset when itemHeight is fixed

### LINKED ISSUE #1538: useVirtualList里的overscan属性设置为0时，第一条数据不会渲染

### PR DESCRIPTION
**Pull request by leedawn**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix: [#1538](https://github.com/alibaba/hooks/issues/1538)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by liuyib** [comment]
@hchlq pls review

File: packages/hooks/src/useVirtualList/index.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ const useVirtualList = <T = any>(list: T[], options: Options<T>) => {
 
   const getOffset = (scrollTop: number) => {
     if (isNumber(itemHeightRef.current)) {
-      return Math.floor(scrollTop / itemHeightRef.current) + 1;
```
Comment (hchlq) [inline packages/hooks/src/useVirtualList/index.ts:54]: 直接去掉的话，这里不会 break change 吗？

Comment (leedawn) [reply packages/hooks/src/useVirtualList/index.ts:54]: 不会的。这里解决的问题是 overscan 为0的时候，getOffset 结果偏大的问题。具体来说就是 itemHeight 为固定值的时候计算结果多加了1，而 itemHeight 为动态值的之后没有问题。另外测试用例没有覆盖到这种情况，所以参考第三个测试用例进行了补充。

**Review by hchlq — APPROVED** [review]

**Comment by hchlq** [comment]
@crazylxr 我这边没问题了哈，辛苦看下之前加上 1 是处理什么问题吗，现在去掉会不会有啥影响

**Pull request merged**